enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,444 @@
+configure the subnets for each availability zone.
+
+## Motivation
+
+Cluster admins on AWS may have dedicated subnets for their load balancers due to
+security reasons, architecture, or infrastructure constraints. Currently, cluster
+admins can manually add the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+annotation to the service. However, this approach is considered bad practice since
+the service is managed by the Ingress Operator. Additionally, the annotation must
+be added after the LoadBalancer-type service has already been created by the Ingress
+Operator. This means that an Ingress Controller may initially be deployed in a broken
+state. Cluster admins need an API for the Ingress Controller which will populate this
+annotation simultaneously with the creation of LoadBalancer-type Services.
+
+### User Stories
+
+#### Day 2 Load Balancer Subnet Selection on AWS
+
+_"As a cluster admin, when configuring an IngressController in AWS, I want to
+specify the subnet selection for the LoadBalancer-type Service on Day 2."_
+
+The cluster admin can modify the `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+field on the IngressController to specify a list of subnets, one for each
+availability zone, for the LoadBalancer-type Service. This is available as a
+Day-2 operation to the cluster admin.
+
+#### Load Balancer Subnet Selection Migration on AWS
+
+_"As a cluster admin, I want to migrate an IngressController LoadBalancer-type
+service, currently using an unmanaged service.beta.kubernetes.io/aws-load-balancer-subnets
+annotation, to using the new IngressController API field so that the annotation is now
+managed."_
+
+A cluster admin manually set the `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation,
+and now wishes to upgrade to OpenShift v4.y to use new IngressController API field,
+`spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`.
+
+1. Cluster admin confirms that initially `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   is set on the LoadBalancer-type service managed by the IngressController.
+2. The cluster admin upgrades OpenShift to v4.y and the annotation is not
+   changed or removed.
+3. From now on, the cluster admin uses `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+   to adjust the ingress controller subnets.
+
+#### Day 1 Load Balancer Subnet Selection on AWS
+
+_"As a cluster admin, when installing a cluster in AWS, I want to specify the subnet
+selection for the default Ingress Controller's load balancer."_
+
+TODO: List steps for this user story
+
+### Goals
+
+- Introduce a new field in the IngressController API for subnet selection in AWS.
+- Support an install-time (Day 1) configuration for subnet selection of the default
+  ingress controller.
+- Support migration from an unmanaged `service.beta.kubernetes.io/aws-load-balancer-subnets`
+  annotation to using the new API field.
```

Comment:
Please state up front (here, and perhaps in the summary as well) that the option will work for both classic ELBs and NLBs.  

- Support the new API field for both classic ELBs and NLBs.